site/source/cookbook/routing/render-widget-url.md

```diff
@@ -0,0 +1,64 @@
+---
+layout: docs
+category: widgets
+title: Render URL aware widgets
+overview: Render a widget when it matches a particular URL
+---
+
+## Objective
+
+This recipe demonstrates how to render one or many widgets based on a URL match.
+
+## Procedure
+
+1. Add the necessary imports in your `main` file:
```

Comment:
I try to avoid use of the word file... either main.ts or just main

Reply:
For these sorts of feedback, I wouldn't say removing the word file is the best approach:

- I'm sure I've read in a technical writing book that you should always specify what the user needs to operate on (e.g. file, method, directory) - but I don't remember where I read it!
- In reading the tutorials, one thing which confused me (and I left feedback on) is when things said "Update the main", I didn't understand what I should be updating
- Even with the extension, it still takes away a sense of "closure" (I know that sounds ridiculous 😃 ), I mean, the sentence feels (to me) incomplete if you don't specify explicitly what the user should be operating on
- Technical writing should be unambiguous and omitting small details like that can introduce a level of ambiguity

Reply:
My main reasoning is that it's not the fact that it is in a file that is important, but rather that it is in a module or class. My thinking was that if someone uses the web-editor, they're not necessarily creating a file in the traditional sense. But perhaps I'm just being silly here.

Reply:
Oh I see the reasoning 👍 If you check out the web editor it still has a tree view:

![image](https://user-images.githubusercontent.com/381495/33259744-c2041828-d355-11e7-93b9-de82179804e7.png)

And I think people would still associate each item in that tree view as a 'file' rather than a 'resource'. How about we try `file` for now?

Reply:
That's fine